src/main/java/net/atos/zac/app/informatieobjecten/InformatieObjectenRESTService.java

```diff
@@ -316,20 +315,35 @@ public Response readFile(@PathParam("uuid") final UUID uuid, @PathParam("versie"
         }
     }
 
-    @GET
-    @Path("/informatieobject/download/zip")
-    @Produces(MediaType.APPLICATION_OCTET_STREAM)
-    public Response readFilesAsZip(@QueryParam("uuids") final String uuids) {
-        final List<String> enkelvoudigInformatieObjectUUIDS = Arrays.stream(uuids.split(",")).toList();
-
+    @POST
+    @Path("/download/zip")
+    public Response readFilesAsZip(final List<String> uuids) {
+        uuids.forEach(uuid -> {
+            UUID enkelvoudigInformatieObjectUUID = UUID.fromString(uuid);
+            assertActie(policyService.readEnkelvoudigInformatieobjectActies(enkelvoudigInformatieObjectUUID).getDownloaden());
+        });
         final StreamingOutput streamingOutput = outputStream -> {
             try (ZipOutputStream zipOutputStream = new ZipOutputStream(new BufferedOutputStream(outputStream))) {
-                enkelvoudigInformatieObjectUUIDS.forEach(stringUuid -> {
+                uuids.forEach(stringUuid -> {
                     final UUID uuid = UUID.fromString(stringUuid);
                     final EnkelvoudigInformatieobject enkelvoudigInformatieobject = drcClientService.readEnkelvoudigInformatieobject(
                             uuid);
                     final ByteArrayInputStream inhoud = drcClientService.downloadEnkelvoudigInformatieobject(uuid);
-                    final ZipEntry zipEntry = new ZipEntry(enkelvoudigInformatieobject.getBestandsnaam());
+                    final URI zaakUri = zrcClientService.listZaakinformatieobjecten(enkelvoudigInformatieobject).get(0).getZaak();
```

Comment:
Maak dit even null safe voor het geval we zip download willen toepassen op ontkoppelde documenten.

Reply:
Opgelost in nieuwe commit